sql/core/src/test/resources/sql-tests/inputs/string-functions.sql

```diff
@@ -24,3 +24,13 @@ select right(null, -2), right("abcd", -2), right("abcd", 0), right("abcd", 'a');
 -- split function
 SELECT split('aa1cc2ee3', '[1-9]+');
 SELECT split('aa1cc2ee3', '[1-9]+', 2);
+
+--substring function
+SELECT substring('abcdefg', 1);
+SELECT substring('abcdefg', -1, 5);
+SELECT substr('abcdefg', 1);
+SELECT substr('abcdefg', -1, 5);
+SELECT substring('abcdefg' from 1);
+SELECT substring('abcdefg' from -1 for 5);
+SELECT substr('abcdefg' from 1);
+SELECT substr('abcdefg' from -1 for 5);
```

Comment:
Could you change the test case? We can have a better test case for the parameters of `substring`. Please use the example from the `substring` function description.
```
    Examples:
      > SELECT _FUNC_('Spark SQL', 5);
       k SQL
      > SELECT _FUNC_('Spark SQL', -3);
       SQL
      > SELECT _FUNC_('Spark SQL', 5, 1);
       k
```

Reply:
I mean the index, 5, -3, (5, 1).